### PR TITLE
PPTP-613: added organisation type pages

### PR DESCRIPTION
**Pull request by tombart**

### CODE REVIEW

**Review by tmc-ee — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/OrganisationType.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  def form(): Form[OrganisationType] =
+    Form(
+      mapping(
+        "answer" -> text()
+          .verifying(emptyError, contains(OrgType.values.toSeq.map(_.toString)))
+      )(OrganisationType.apply)(OrganisationType.unapply)
+    )
+
+  def apply(value: String): OrganisationType = OrganisationType(OrgType.withNameOpt(value).orNull)
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/OrganisationType.scala:47]: `Null` in Scala is a smell.

This was my approach https://github.com/hmrc/plastic-packaging-tax-registration-frontend/pull/67/files#diff-eb502f07a3e7e1fefe9fec6789846c27d34ef8a77ddcc1176b960c66b0c63313R21-R28

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package uk.gov.hmrc.plasticpackagingtax.registration.models.registration
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.{Address, OrgType}
+
+case class OrganisationDetails(
+  isBasedInUk: Option[Boolean] = None,
+  organisationType: Option[OrgType.Value] = None,
+  businessRegisteredAddress: Option[Address] = None
+)
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:26]: Good approach!! Well organised!

File: conf/messages.en [hunk 3]
```diff
@@ -167,7 +167,7 @@ reviewRegistration.sendYourApplication.title = Now send your application
 reviewRegistration.sendYourApplication.body = By submitting this registration you are confirming that, to the best of your knowledge, the details you are providing are correct.
 
 confirmationPage.title = Registration submitted
-confirmationPage.panel.body = Your reference number {0}
+confirmationPage.panel.body = Your reference numfeedbackber {0}
```
Comment (PauloPortugal) [inline conf/messages.en:170]: This is probably a typo

**Review by PauloPortugal — CHANGES_REQUESTED** [review]
There is a typo on the `messages.en`. Please correct this.

Apart from that all looks good. I would change the approach of how we are creating the Enumeration to reflect a more Scala approach, so that we avoid null references in the code.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/OrganisationType.scala (hunk 1 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/OrganisationType.scala]: done

File: conf/messages.en (hunk 3 above)
Comment (tombart) [reply conf/messages.en]: fixed

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**